pkg/github/synchronizer.go

```diff
@@ -43,42 +43,46 @@ func NewSynchronizer(ghClient *github.Client, ghApp *githubauth.App) *Synchroniz
 	}
 }
 
-// Sync overides a GitHub team's memberships with the provided team membership
-// snapshot.
+// Sync overides several GitHub teams' memberships with the provided team
+// membership snapshots.
 // TODO(#3): populate the users' GitHub logins in the GitHubTeam object before
 // this since they are required when updating GitHub team memberships.
-func (s *Synchronizer) Sync(ctx context.Context, team *v1alpha1.GitHubTeam) error {
+func (s *Synchronizer) Sync(ctx context.Context, teams []*v1alpha1.GitHubTeam) error {
 	// Configure Github auth token to the GitHub client.
 	t, err := s.accessToken(ctx)
 	if err != nil {
 		return fmt.Errorf("failed to get access token: %w", err)
 	}
 	ghClient := s.client.WithAuthToken(t)
 
-	// Get current team members' login from GitHub and expected team members'
-	// user from the team object.
-	gotActiveMemberLogins, err := s.currentTeamLogins(ctx, ghClient, team.GetOrgId(), team.GetTeamId(), listActiveTeamMembers)
-	if err != nil {
-		return fmt.Errorf("failed to get active GitHub team members: %w", err)
-	}
-	gotPendingInvitationLogins, err := s.currentTeamLogins(ctx, ghClient, team.GetOrgId(), team.GetTeamId(), listPendingTeamInvitations)
-	if err != nil {
-		return fmt.Errorf("failed to get pending GitHub team invitations: %w", err)
-	}
-	gotLogins := sets.Union(gotActiveMemberLogins, gotPendingInvitationLogins)
-	wantLogins := loginsFromTeam(team)
-
 	var retErr error
-	// Add GitHub team memberships.
-	for _, u := range sets.Subtract(wantLogins, gotLogins) {
-		if _, _, err := ghClient.Teams.AddTeamMembershipByID(ctx, team.GetOrgId(), team.GetTeamId(), u, &github.TeamAddTeamMembershipOptions{Role: "member"}); err != nil {
-			retErr = errors.Join(retErr, fmt.Errorf("failed to add GitHub team members: %w", err))
+	for _, team := range teams {
+		// Get current team members' login from GitHub and expected team members'
+		// user from the team object.
+		gotActiveMemberLogins, err := s.currentTeamLogins(ctx, ghClient, team.GetOrgId(), team.GetTeamId(), listActiveTeamMembers)
+		if err != nil {
+			retErr = errors.Join(retErr, fmt.Errorf("failed to get active GitHub team members for team(%d): %w", team.GetTeamId(), err))
+			continue
```

Comment:
Do you need continue here? In the PR, for the first two for loops, you use continue, and for the other two loops, you don't use continue.

Reply:
That is intentional, so that when get the current team's membership and invitation (first 2 loops) failed, we don't update memberships for the team, but if there are errors when updating a member's membership to a team, we can still update other members.